pkg/k8s/apis/cilium.io/v2/validator/validator.go

```diff
@@ -105,10 +109,89 @@ func (n *NPValidator) ValidateCNP(cnp *unstructured.Unstructured) error {
 	return nil
 }
 
+const (
+	// We can remove the check for this warning once 1.9 is the oldest supported Cilium version.
+	warnEmptyToFromEndpointMessage = "It seems you have a CiliumClusterwideNetworkPolicy " +
+		"with an empty to/from endpoint selector. The behavior of this selector has been " +
+		"changed. The selector now only allows traffic to/from Cilium managed K8s endpoints, " +
+		"instead of acting as a truly empty endpoint selector allowing all traffic. To " +
+		"ensure that the policy behavior does not affect your workloads, consider adding " +
+		"another policy that allows traffic to/from world and cluster entities. For a more " +
+		"detailed discussion on the topic, see GitHub issue #12844"
+)
+
 // ValidateCCNP validates the given CCNP accordingly the CCNP validation schema.
 func (n *NPValidator) ValidateCCNP(ccnp *unstructured.Unstructured) error {
 	if errs := validation.ValidateCustomResource(nil, &ccnp, n.ccnpValidator); len(errs) > 0 {
 		return errs.ToAggregate()
 	}
+
+	logger := log.WithFields(logrus.Fields{
+		logfields.CiliumClusterwideNetworkPolicyName: ccnp.GetName(),
+	})
+
+	// At this point we have validated the custom resource with the new CRV.
+	// We can try converting it to the new CCNP type.
+	// This should not fail, so we are not returning any errors, just logging
+	// a warning.
+	ccnpBytes, err := ccnp.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	resCCNP := cilium_v2.CiliumClusterwideNetworkPolicy{}
+	err = json.Unmarshal(ccnpBytes, &resCCNP)
+	if err != nil {
+		return err
+	}
+
+	// Print the warninig only once per CCNP.
```

Comment:
I'm a little concerned about how frequently this log message could be printed, surely it's not more useful than once per Cilium run? ie wrap with `sync.Once()`?

Reply:
Yeah, I think it makes sense to print the warning only once to hint the users about the changes. I will update the pull request.